### PR TITLE
Moremetrics

### PR DESCRIPTION
**Pull request by MCBrandenburg**
Hello, due to the needs of my company I've added the following metrics to this project:

`solidfire_cluster_account_count`
`solidfire_cluster_admin_account`
`solidfire_cluster_initiator_count`
`solidfire_cluster_volume_access_group_count`

### CODE REVIEW

File: pkg/prom/metrics.go [hunk 1]
```diff
@@ -926,5 +933,40 @@ func NewMetricDescriptions(namespace string) *Descriptions {
...
+	d.VolumeCount = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "cluster_volume_count"),
+		"The total number of volumes in cluster",
+		nil,
+		nil,
+	)
+
+	d.AccountCount = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "cluster_account_count"),
+		"The total number of acocunts in cluster",
```
Comment (willgorman) [inline pkg/prom/metrics.go:945]:
```suggestion
		"The total number of accounts in cluster",
```

**Comment by neufeldtech** [comment]
Hey@MCBrandenburg ,

Can you share a little more about your use case for the count of volumes? You should be able to get this info already from using the Prometheus count() function on any of the other volume related metrics.

**Comment by MCBrandenburg** [comment]
@neufeldtech good catch, I've updated the pr to remove volume count

**Comment by neufeldtech** [comment]
@MCBrandenburg ,

Before we get too far into the review, I'd like to propose an alternative strategy. As per the guidance at [Metric and Label Naming Guidance](https://prometheus.io/docs/practices/naming/#metric-and-label-naming), we can refactor the `account`, `admin_account`, `initiator`, and `volume_access_group` metrics to be pseudo metrics with the `_info` suffix and expose their gauges with the static value of 1.

We've done a similar strategy for the `solidfire_node_info` metric as it's a similar type where the number itself isn't really of interest, but the interesting parts are in the labels exposed.

```
solidfire_node_info{associated_fservice_id="0",associated_master_service_id="1",chassis_name="",chassis_type="SFVIRT",cpu_model="Intel(R) Xeon(R) CPU E7-8891 v4 @ 2.80GHz\nUnknown Processor",node_id="1",node_name="n01",node_type="SFDEMO-NE",platform_config_version="0.0.0.0",sip="10.0.0.91",sipi="eth1",software_version="11.7.0.76",uuid="5329FE1F-A41F-DC41-8BD9-2016FF2DD8FF"} 1
```

I'm envisioning something like this:

```
# For these metrics specifically, we will likely want to exclude secrets labels from being exposed here
solidfire_cluster_account_info{account_id="16",status="active",storageContainerID="abcdef01-1234-5678-90ab-cdef01234567".....} 1
```

```
# For this one, there may be multiple strategies for how to expose the list of access roles granted to an account, below seems to be one option. 
solidfire_cluster_admin_account_info{cluster_admin_id="1",username="admin",access="administrator,read"} 1
```

```
# For initiator info, is `cluster` the correct namespace? Or is it better served in its own metric namespace (i.e. solidfire_initiator_info) ?
# Also, there are multiple options here for how we could expose the volume access groups that are bound to an initiator... Below is an option where we have one metric per volume_access_group that this initiator has access to, but we could consider other ways of doing this too. I'm not super familiar with this domain (it's been a while), so I don't have insight into how this data may want to be used after it's collected.

solidfire_cluster_initiator_info{initiator_id="2",initiator_name="iqn.1993-08.org.debian:01:c84ffd71216",volume_access_group="1"} 1
solidfire_cluster_initiator_info{initiator_id="2",initiator_name="iqn.1993-08.org.debian:01:c84ffd71216",volume_access_group="2"} 1
```

```
# For this one, I propose dropping the `cluster`, as this looks like a volume specific metric
# This one has a number of lists in the api response... another option of potentially dealing with this kind of multi-faceted data is to include counts of relevant info as a label, as in number of deleted volumes, initiators, etc...
solidfire_volume_access_group_info{volume_access_group_id="1",name="myvolume",initiators="3",deleted_volumes="4"}
```

**Comment by MCBrandenburg** [comment]
@neufeldtech thank you for the input. I'm going to close this and revisit later on when I'm able to implement.

**Pull request closed without merging**